apps/files/lib/Dashboard/FavouriteWidget.php

```diff
@@ -0,0 +1,157 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * SPDX-FileCopyrightText: 2020 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+namespace OCA\Files\Dashboard;
+
+use OCA\Files\AppInfo\Application;
+use OCP\Dashboard\IAPIWidget;
+use OCP\Dashboard\IAPIWidgetV2;
+use OCP\Dashboard\IButtonWidget;
+use OCP\Dashboard\IIconWidget;
+use OCP\Dashboard\IWidget;
+use OCP\Dashboard\Model\WidgetButton;
+use OCP\Dashboard\Model\WidgetItem;
+use OCP\Dashboard\Model\WidgetItems;
+use OCP\Files\IMimeTypeDetector;
+use OCP\Files\IRootFolder;
+use OCP\IL10N;
+use OCP\IPreview;
+use OCP\ITagManager;
+use OCP\IURLGenerator;
+use OCP\IUserManager;
+use OCP\IUserSession;
+
+class FavouriteWidget implements IWidget, IIconWidget, IAPIWidget, IAPIWidgetV2, IButtonWidget {
+	private IUserSession $userSession;
+	private IL10N $l10n;
+	private IURLGenerator $urlGenerator;
+	private IMimeTypeDetector $mimeTypeDetector;
+	private IUserManager $userManager;
+	private ITagManager $tagManager;
+	private IRootFolder $rootFolder;
+	private IPreview $previewManager;
+	public const FAVORITE_LIMIT = 50;
+
+	public function __construct(
+		IUserSession $userSession,
+		IL10N $l10n,
+		IURLGenerator $urlGenerator,
+		IMimeTypeDetector $mimeTypeDetector,
+		IUserManager $userManager,
+		ITagManager $tagManager,
+		IRootFolder $rootFolder,
+		IPreview $previewManager,
+	) {
+		$this->userSession = $userSession;
+		$this->l10n = $l10n;
+		$this->urlGenerator = $urlGenerator;
+		$this->mimeTypeDetector = $mimeTypeDetector;
+		$this->userManager = $userManager;
+		$this->tagManager = $tagManager;
+		$this->rootFolder = $rootFolder;
+		$this->previewManager = $previewManager;
+	}
```

Comment:
```suggestion
	public function __construct(
		private IIUserSession $userSession,
		private IIL10N $l10n,
		private IIURLGenerator $urlGenerator,
		private IIMimeTypeDetector $mimeTypeDetector,
		private IIUserManager $userManager,
		private IITagManager $tagManager,
		private IIRootFolder $rootFolder,
		private IIPreview $previewManager,
	) {
	}
```
Use promoted properties on new classes.